overcooked_ai_py/planning/planners.py

```diff
@@ -811,19 +811,27 @@ def get_medium_level_actions(self, state, player, waiting_substitute=False):
         player_actions = list(filter(is_valid_goal_given_start, player_actions))
         return player_actions
 
-    def pickup_onion_actions(self, state, counter_objects):
-        onion_dispenser_locations = self.mdp.get_onion_dispenser_locations()
-        onion_pickup_locations = onion_dispenser_locations + counter_objects['onion']
+    def pickup_onion_actions(self, state, counter_objects, only_use_dispensers=False):
+        """If only_use_dispensers is True, then only take onions from the dispensers"""
+        if not only_use_dispensers:
+            onion_dispenser_locations = self.mdp.get_onion_dispenser_locations()
+            onion_pickup_locations = onion_dispenser_locations + counter_objects['onion']
+        else:
+            onion_pickup_locations = self.mdp.get_onion_dispenser_locations()
```

Comment:
Don't need to repeat the `self.mdp.get_onion_dispenser_location()` call twice, can place it before the `if` statement and have no `else`.